src/slurm_plugin/clustermgtd.py

```diff
@@ -964,6 +973,73 @@ def _is_node_in_replacement_valid(self, node, check_node_is_valid):
             return not time_is_expired if check_node_is_valid else time_is_expired
         return False
 
+    def _handle_ice_nodes(self, active_nodes, unhealthy_dynamic_nodes):
+        ice_nodes = [node for node in unhealthy_dynamic_nodes if node.is_ice()]
+        self.update_insufficient_capacity_compute_resources()
+        self.reset_timeout_expire_compute_resources(ice_nodes)
+        self._set_ice_crs_to_down(active_nodes)
+
+    def update_insufficient_capacity_compute_resources(self, ice_nodes):
```

Comment:
nit: mind the convention for "private" methods

Reply:
Changed to _update_insufficient_capacity_compute_resources